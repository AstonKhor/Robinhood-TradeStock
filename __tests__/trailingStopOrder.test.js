/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TrailingStopOrder from '../client/components/trailingStopOrder.jsx';

describe('<TrailingStopOrder /> Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<TrailingStopOrder />);
    expect(tree).toMatchSnapshot();
  });

  it('initiates state as expected', () => {
    const wrapper = shallow(<TrailingStopOrder />);
    expect(wrapper.state().trail).toBe(0);
    expect(wrapper.state().trailType).toBe('Percentage');
    expect(wrapper.state().shares).toBe(0);
    expect(wrapper.state().expires).toBe('Good for Day');
  });
});
