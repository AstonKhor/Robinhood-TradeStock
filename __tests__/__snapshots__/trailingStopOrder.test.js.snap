// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrailingStopOrder /> Component matches the snapshot 1`] = `
Array [
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Trail Type
    </label>
    <div
      className="sc-htoDjs gmNRED"
      onClick={[Function]}
    >
      Percentage
    </div>
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Trail (
      %
      )
    </label>
    <input
      className="sc-bdVaJa sc-htpNat dFFWgj"
      name="trail"
      onChange={[Function]}
      step=".01"
      type="number"
      value={0}
    />
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Shares
    </label>
    <input
      className="sc-bdVaJa sc-htpNat dFFWgj"
      name="shares"
      onChange={[Function]}
      placeholder="0"
      type="number"
      value={0}
    />
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Expires
    </label>
    <div
      className="sc-htoDjs gmNRED"
      onClick={[Function]}
    >
      Good for Day
    </div>
  </div>,
]
`;
